src/tovnode.ts

```diff
@@ -2,6 +2,18 @@ import { addNS } from "./h";
 import { vnode, VNode } from "./vnode";
 import { htmlDomApi, DOMAPI } from "./htmldomapi";
 
+/**
+ * Transforms the given attribute name into a valid dataset property key
+ * according to https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#name_conversion
+ *
+ * @param attributeName data- attribute name, must start with data-
+ */
+function datasetKey(attributeName: string) {
+  return attributeName
+    .slice(5)
```

Comment:
```suggestion
    .toLowerCase()
    .slice(5)
```

Just to be inline with https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#name_conversion

Reply:
We can not use `toLowerCase` as that does not match the MDN description. It states that only `A-Z` should be converted to lowercase, but `toLowerCase` also handles unicode. For instance `"Æ".toLowerCase()` gives `æ`. If we want to be in line with the spec I think using a regex would be easiest.

Reply:
It seems that lowercasing A-Z characters is also done at the [HTML level](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#in_html). So when reading elements through the DOM we can assume that there are no upper case A-Z characters. The current implementation should therefore be fine :+1: